app/scripts/museum/museum-service.js

```diff
@@ -0,0 +1,66 @@
+
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function Museum ($http, Config) {
+        var sqlApi = ['https://', Config.cartodb.account,
+                      '.cartodb.com/api/v2/sql'].join('');
+        var cols = {
+            id: 'cartodb_id',
+            name: 'commonname',
+            altname: 'altname',
+            legalname: 'legalname',
+            geom: 'the_geom'
+        };
+
+        var module = {
+            sqlQuery: sqlQuery,
+            suggest: suggest
+        };
+
+        return module;
+
+        // Hoisted function definitions
+
+        function sqlQuery(query) {
+            return $http.get(sqlApi, {
+                params: {
+                    q: query
+                }
+            });
+        }
+
+        function escapeString(value, type) {
+            if (!type) {
+                type = 'text';
+            }
+            return "convert_from(decode('" + window.btoa(value) +
+                   "','base64'), getdatabaseencoding())::"+type;
+        }
+
+        function suggest(text) {
+            var query = ['select ',
```

Comment:
Why not use `cartodb.SQL` for this? It handles most of this stuff for you:
http://docs.cartodb.com/cartodb-platform/cartodb-js.html#cartodbsql

Strings are escaped internally by cartodb and windshaft AFAIK